### PR TITLE
Text renderer

### PR DESCRIPTION
**Pull request by athaun**
image

I would greatly appreciate a thorough review of my code, and I do realize it is quite alot of code to review.
From what I can see, git is confused about some files, and has marked them as completely changed, which makes it much more difficult to review.
The files to focus your review on are as follows
* graphics/renderer/TextRenderer.java
* graphics/renderer/TextRendererBatch.java
* fonts/GlyphRenderer.java
* fonts/Glyph.java
* fonts/Font.java
* ecs/Text.java
* scene/Scene.java
* scenes/TextRenderingDemo.java

Notable changes in this commit are:
* Font renderer
* Additional documentation and author attribution
* Beginning to lay the groundwork for further UI.
* Disabling sprite rotation for now until we can come up with a cleaner solution (as discussed on discord)
* deprecating the DemoPlatformer scene - it needs to either be rewritten or removed before 1.0.

### CODE REVIEW

File: src/main/java/scene/Scene.java [hunk 1]
```diff
@@ -1,315 +1,338 @@
...
+     */
+    public void initRenderers() {
+        debugRenderer.init();
+        lightmapRenderer.init();
+        renderer.init();
+        forwardToScreen = new ForwardToTexture(PostProcessStep.Target.DEFAULT_FRAMEBUFFER);
+        forwardToScreen.init();
+    }
+
+    public void addUiObject (Text t) {
```
Comment (Juyas) [inline src/main/java/scene/Scene.java:301]: We might change this method, if UI is designed.

File: src/main/java/ecs/GameObject.java [hunk 2]
```diff
@@ -1,273 +1,258 @@
...
+     * @param zIndex
+     */
+    public GameObject(String name, List<Component> componentList, Vector3f locationData, int zIndex) {
+        this.name = name;
+        this.components = new OrderPreservingList<>(componentList);
+        this.locationData = locationData;
+        this.locationBuffer = new Vector3f();
+        this.zIndex = zIndex;
+        this.parentScene = Engine.window().currentScene();
+        Engine.window().currentScene().addGameObjectToScene(this);
```
Comment (Juyas) [inline src/main/java/ecs/GameObject.java:54]: This is a hidden dependency we removed ealier... I am not sure, if we should do it like that.

**Comment by Juyas** [comment]
I would approve of this, maybe add some structure tests to polish a little.

File: src/main/java/scene/Scene.java (hunk 1 above)
Comment (athaun) [reply src/main/java/scene/Scene.java:301]: yes, agreed.

File: src/main/java/ecs/GameObject.java (hunk 2 above)
Comment (athaun) [reply src/main/java/ecs/GameObject.java:54]: I am not sure what you mean by hidden, it is a publicly accessible method.  
Personally I dislike having to pass the Scene, but we could compromise with an additional constructor.

Comment (Juyas) [reply src/main/java/ecs/GameObject.java:54]: we already did compromise on that in discord, so pls add it ;)

Comment (athaun) [reply src/main/java/ecs/GameObject.java:54]: This concern is now resolved.

**Pull request merged**